run/run.go

```diff
@@ -1,47 +1,154 @@
 	"github.com/Portshift/klar/config"
 	"github.com/Portshift/klar/docker"
-	log "github.com/sirupsen/logrus"
+	anchore_image "github.com/anchore/stereoscope/pkg/image"
+	"github.com/anchore/syft/syft"
+	"github.com/anchore/syft/syft/format"
+	"github.com/anchore/syft/syft/source"
+	httptransport "github.com/go-openapi/runtime/client"
+	"github.com/go-openapi/strfmt"
 )
 
-func ExecuteScan(conf *config.Config) ([]*clair.Vulnerability, []*docker.FsLayerCommand, error) {
-	image, err := docker.NewImage(&conf.DockerConfig)
+func ExecuteRemoteGrypeScan(imageName string, conf *config.Config) (*grype_models.Document, []*docker.FsLayerCommand, error) {
+	src, cleanup, err := source.New(imageName, &anchore_image.RegistryOptions{
+		InsecureSkipTLSVerify: conf.DockerConfig.InsecureTLS,
+		InsecureUseHTTP:       conf.DockerConfig.InsecureRegistry,
+		Credentials: []anchore_image.RegistryCredentials{
+			{
+				Authority: "", // What is this?
+				Username:  conf.DockerConfig.User,
+				Password:  conf.DockerConfig.Password,
+				Token:     conf.DockerConfig.Token,
+			},
+		},
+	})
 	if err != nil {
-		return nil, nil, fmt.Errorf("failed to parse name: %v", err)
+		return nil, nil, fmt.Errorf("failed to create syft source: %v", err)
 	}
+	defer cleanup()
 
-	err = image.Pull()
+	catalog, distro, err := syft.CatalogPackages(src, source.SquashedScope)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to catalog packages: %v", err)
+	}
+	sbomEncoded, err := syft.Encode(catalog, &src.Metadata, distro, source.SquashedScope, format.JSONOption)
 	if err != nil {
-		return nil, nil, fmt.Errorf("failed to pull image: %w", err)
+		return nil, nil, fmt.Errorf("failed to encode sbom: %v", err)
 	}
 
-	if err := image.FetchFsCommands(&conf.DockerConfig); err != nil {
-		return nil, nil, fmt.Errorf("failed to fetch layer commands: %v", err)
+	sbom64 := base64.StdEncoding.EncodeToString(sbomEncoded)
+
+	client := createGrypeClient(conf.GrypeAddr)
+	params := grype_client_operations.NewPostScanSBOMParams().WithBody(&models.SBOM{
+		Sbom: sbom64,
+	})
+	ok, err := client.Operations.PostScanSBOM(params)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to send sbom for scan: %v", err)
+	}
+	doc := grype_models.Document{}
+	docB, err := base64.StdEncoding.DecodeString(ok.Payload.Vulnerabilities)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to decode vulnerabilities: %v", err)
+	}
+	err = json.Unmarshal(docB, &doc)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to unmarshall vulnerabilities document: %v", err)
 	}
 
-	if len(image.FsLayers) == 0 {
-		return nil, nil, fmt.Errorf("failed to pull pull fsLayers")
+	commands, err := GetImageCommands(conf)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to get image commands : %v", err)
+	}
+
+	return &doc, commands, nil
+}
+
+func ExecuteLocalGrypeScan(imageName string, conf *config.Config) (*grype_models.Document, []*docker.FsLayerCommand, error) {
```

Comment:
can you please add a comment above ExecuteLocalGrypeScan and ExecuteRemoteGrypeScan to describe shortly what each function does?